### ISSUE TITLE
Inithooks/Confconsole display garbled content with PuTTY

### ISSUE THREAD

**Issue by JedMeister** [turnkeylinux/tracker#799]
It is not likely that we will be able to provide a proper "fix" for this, but ideally we should at least document workarounds.

The issue appears to be related to the way that SSH server sets the character encoding and the way that Windows & PuTTY interpret that. 

It can in theory be worked around server side, but initial testing has suggested that applying any of those workarounds server side (i.e. by default within TurnKey builds) doesn't work consistently across Windows PCs. In many cases it can cause the same problem within terminal SSH-client connections on Mac and Linux (i.e. fixing it for some Windows users, often breaks it for Mac & Linux users).

It can be worked around server side by setting the server locale appropriately and/or changing the `TERM` environment variable.

It can be worked around client side by tweaking the PuTTY character encoding. More testing is required and proper explicit workaround(s) posted.

FWIW there are a number of potential workarounds noted in [this ServerFault Q&A](https://serverfault.com/questions/475925/how-to-fix-putty-showing-garbled-characters). Please note though that we have had report that simply setting PuTTY to use UTF-8 is not sufficient.

**Comment by Dude4Linux** [comment]
@JedMeister I think I'm seeing this issue while trying to capture screenshots of confconsole via a remote ssh login from Ubuntu (no PuTTY involved). The procedure works as expected with the 14.1 confconsole version 1.0.0+4+g18a, but I get the garbled output from version 1.0.1. This is for a new appliance built for 14.2. I've examined both host configs and the only difference seems to be the version of confconsole.

![confconsole_usage_snap](https://user-images.githubusercontent.com/1219334/27297865-3175e4cc-54ec-11e7-824b-27f2ac3e018a.jpg)

**Comment by Dude4Linux** [comment]
@JedMeister Nevermind, I checked a different 14.2 appliance and don't see the problem there. Apparently I picked up a newer version of the ncurses libraries when I installed the latest OpenVAS.  Will have to see if I can downgrade them.

**Comment by JedMeister** [comment]
Yeah we really need to update that (we should ideally be depending on the version of python-dialog from the Debian repos). In fact most of the TurnKey specific code needs serious update. It's been in need of update for some time...

But as per most non-urgent stuff, I'm not really sure when we're going to get there. We need to finish up v14.2. Then we need to get v15.0 out the door (Stretch was released last weekend). So I don't see it happening before v15.0 is completed...

**Comment by JedMeister** [comment]
Whilst it appears that we can't "fix" this by default, we _should_ properly document how to fix it!

**Comment by JedMeister** [comment]
Bumping to next milestone...

**Comment by huzhengming** [comment]
> @JedMeister I think I'm seeing this issue while trying to capture screenshots of confconsole via a remote ssh login from Ubuntu (no PuTTY involved). The procedure works as expected with the 14.1 confconsole version 1.0.0+4+g18a, but I get the garbled output from version 1.0.1. This is for a new appliance built for 14.2. I've examined both host configs and the only difference seems to be the version of confconsole.
> 
> ![confconsole_usage_snap](https://user-images.githubusercontent.com/1219334/27297865-3175e4cc-54ec-11e7-824b-27f2ac3e018a.jpg)

export TERM= linux

**Comment by huzhengming** [comment]
终端输入export TERM= linux  既可以解决

**Comment by huzhengming** [comment]
终端输入export TERM= linux 既可以解决

**Comment by huzhengming** [comment]
export TERM= linux

**Comment by OnGle** [comment]
Can we get somebody else from a non-linux environment to test this workaround?

**Comment by JedMeister** [comment]
I'm not completely sure whether this is still an issue or not? Can a non-linux user confirm that this is no longer an issue?

**Comment by OnGle** [comment]
Closing due to inactivity.

---

But also this is a terminal detection issue, and is not globally fixable:

Virtual terminals communicate which terminal they are (or are emulating) via the `TERM` environment variable , and the active program may query the terminfo database to find out information such as what capabilities they have (which colors they can display, what characters they can input, etc.).

One of these things that is checked is availability of and mappings to block & line drawing glyphs.

See the following table taken from the `terminfo(5)` man page

```

       Glyph                       ACS            Ascii     acsc     acsc
       Name                        Name           Default   Char     Value
       ────────────────────────────────────────────────────────────────────
       arrow pointing right        ACS_RARROW     >         +        0x2b
       arrow pointing left         ACS_LARROW     <         ,        0x2c
       arrow pointing up           ACS_UARROW     ^         -        0x2d
       arrow pointing down         ACS_DARROW     v         .        0x2e
       solid square block          ACS_BLOCK      #         0        0x30
       diamond                     ACS_DIAMOND    +         `        0x60
       checker board (stipple)     ACS_CKBOARD    :         a        0x61
       degree symbol               ACS_DEGREE     \         f        0x66
       plus/minus                  ACS_PLMINUS    #         g        0x67
       board of squares            ACS_BOARD      #         h        0x68
       lantern symbol              ACS_LANTERN    #         i        0x69
       lower right corner          ACS_LRCORNER   +         j        0x6a
       upper right corner          ACS_URCORNER   +         k        0x6b
       upper left corner           ACS_ULCORNER   +         l        0x6c
       lower left corner           ACS_LLCORNER   +         m        0x6d
       large plus or crossover     ACS_PLUS       +         n        0x6e
       scan line 1                 ACS_S1         ~         o        0x6f
       scan line 3                 ACS_S3         -         p        0x70
       horizontal line             ACS_HLINE      -         q        0x71
       scan line 7                 ACS_S7         -         r        0x72
       scan line 9                 ACS_S9         _         s        0x73
       tee pointing right          ACS_LTEE       +         t        0x74
       tee pointing left           ACS_RTEE       +         u        0x75
       tee pointing up             ACS_BTEE       +         v        0x76
       tee pointing down           ACS_TTEE       +         w        0x77
       vertical line               ACS_VLINE      |         x        0x78
       less-than-or-equal-to       ACS_LEQUAL     <         y        0x79
       greater-than-or-equal-to    ACS_GEQUAL     >         z        0x7a
       greek pi                    ACS_PI         *         {        0x7b
       not-equal                   ACS_NEQUAL     !         |        0x7c
       UK pound sign               ACS_STERLING   f         }        0x7d
       bullet                      ACS_BULLET     o         ~        0x7e
```

Given the above table and this snippet:

```
       •   The DEC VT100 implemented graphics using the alternate
           character set feature, temporarily switching modes and sending
           characters in the range 0x60 (96) to 0x7e (126) (the acsc
           Value column in the table).
```

It is clear that either the `TERM` value is being set incorrectly, or the value of the `TERM` variable is not in the `terminfo` database. (Note how values like `q` are mapping to a `horizontal line`)

We cannot manually add terminfo entries for every terminal and maintain them as the capabilities of each terminal change.

The best bet for users facing this issue is to use common and widely supported term values, that many terminals support super-sets of such as:

`xterm`
`xterm-256color`
`linux`
`vt100`

Or just find the closest match you can in the available terminfo definitions, to see them all run:

`find /usr/share/terminfo/ -type f | sed -n 's|.*/\(.*$\)|\1|p' | sort -u`
